test/hummingbot/connector/exchange/kraken/test_kraken_api_order_book_data_source.py

```diff
@@ -0,0 +1,251 @@
+                    "l",
+                    ""
+                ],
+                [
+                    "6060.00000",
+                    "0.02455000",
+                    "1534614057.324998",
+                    "b",
+                    "l",
+                    ""
+                ]
+            ],
+            "trade",
+            f"{self.base_asset}/{self.quote_asset}"
+        ]
+        return trade_data
+
+    @aioresponses()
+    def test_get_last_traded_prices(self, mocked_api):
+        url = f"{CONSTANTS.BASE_URL}{CONSTANTS.TICKER_PATH_URL}"
+        regex_url = re.compile(f"^{url}".replace(".", r"\.").replace("?", r"\?"))
+        last_traded_price = Decimal("52641.10000")
+        resp = self.get_last_traded_prices_mock(last_trade_close=last_traded_price)
+        mocked_api.get(regex_url, body=json.dumps(resp))
+
+        ret = self.async_run_with_timeout(
+            KrakenAPIOrderBookDataSource.get_last_traded_prices(
+                trading_pairs=[self.trading_pair], throttler=self.throttler
+            )
+        )
+
+        self.assertIn(self.trading_pair, ret)
+        self.assertEqual(float(last_traded_price), ret[self.trading_pair])
+
+    @aioresponses()
+    def test_get_new_order_book(self, mocked_api):
+        url = f"{CONSTANTS.BASE_URL}{CONSTANTS.SNAPSHOT_PATH_URL}"
+        regex_url = re.compile(f"^{url}".replace(".", r"\.").replace("?", r"\?"))
+        resp = self.get_depth_mock()
+        mocked_api.get(regex_url, body=json.dumps(resp))
+
+        ret = self.async_run_with_timeout(self.data_source.get_new_order_book(self.trading_pair))
+
+        self.assertTrue(isinstance(ret, OrderBook))
+
+    @aioresponses()
+    def test_fetch_trading_pairs(self, mocked_api):
+        url = f"{CONSTANTS.BASE_URL}{CONSTANTS.ASSET_PAIRS_PATH_URL}"
+        regex_url = re.compile(f"^{url}".replace(".", r"\.").replace("?", r"\?"))
+        resp = self.get_public_asset_pair_mock()
+        mocked_api.get(regex_url, body=json.dumps(resp))
+
+        resp = self.async_run_with_timeout(KrakenAPIOrderBookDataSource.fetch_trading_pairs())
+
+        self.assertTrue(len(resp) == 1)
+        self.assertIn(self.trading_pair, resp)
+
+    @patch("websockets.connect", new_callable=AsyncMock)
+    def test_listen_for_trades(self, ws_connect_mock):
+        ws_connect_mock.return_value = self.mocking_assistant.create_websocket_mock()
+        resp = self.get_trade_data_mock()
+        self.mocking_assistant.add_websocket_text_message(
+            websocket_mock=ws_connect_mock.return_value, message=json.dumps(resp)
+        )
+        output_queue = asyncio.Queue()
+
+        self.ev_loop.create_task(self.data_source.listen_for_trades(self.ev_loop, output_queue))
+        self.mocking_assistant.run_until_all_text_messages_delivered(websocket_mock=ws_connect_mock.return_value)
+
+        self.assertTrue(not output_queue.empty())
+        self.assertTrue(isinstance(output_queue.get_nowait(), OrderBookMessage))
```

Comment:
Here again it would be better to assert something based on the content of the order book message, instead of checking its class.